webapp/src/Service/SubmissionService.php

```diff
@@ -768,4 +768,33 @@ public function getSubmissionZipResponse(Submission $submission): StreamedRespon
 
         return Utils::streamZipFile($tmpfname, 's' . $submission->getSubmitid() . '.zip');
     }
+
+    public function getSubmissionFileResponse(Submission $submission): StreamedResponse
+    {
+        /** @var SubmissionFile[] $files */
+        $files = $submission->getFiles();
+        
+        if (count($files) !== 1) {
+            throw new ServiceUnavailableHttpException(null, 'Submission does not contain exactly one file.');
+        }
+
+        $file = $files[0];
+        $filename = $file->getFilename();
+        $sourceCode = $file->getSourcecode();
+
+        return new StreamedResponse(function () use ($sourceCode) {
+            echo $sourceCode;
+        }, 200, [
+            'Content-Type'        => 'application/octet-stream',
```

Comment:
I wonder whether this should be `text/plain` or something different?

Reply:
application/octet-stream is more commonly used when downloading files, but I just tested it, and the effect here is the same as text/plain.